File: ios/Sources/TaigiKeyboard/Autocomplete/Services/AutocompleteInputClassifier.swift
// 中文: rawInput 分類器的 Swift 端薄包裝 — 把字串送進 Rust 引擎判斷輸入型別。

import Foundation

/// 依 rawInput 判斷輸入型別並建立 Trie 搜尋鍵。
///
/// Thin wrapper over `RustEngineBridge.classifyInput`. Precedence contract:
/// `INVARIANT_LEX_INPUT_CLASSIFICATION_PRECEDENCE`.
enum AutocompleteInputClassifier {
    // 中文: 分類結果 — 對應的 InputType 加上要丟進 Trie 的 searchKey。
    struct Classification: Equatable {
        let inputType: InputType
        let searchKey: String
    }

    /// `classify` 把 rawInput 轉成 `(inputType, searchKey)` 對。
    /// - rawInput 例：`gua2` / `guá` / `我` / TPS 符號
    static func classify(rawInput: String) -> Classification {
        let result = RustEngineBridge.classifyInput(rawInput)
        return Classification(inputType: result.inputType, searchKey: result.searchKey)
    }
}

File: android/app/src/main/java/com/siansiansu/taigikeyboard/ime/text/composing/AutocompleteInputClassifier.kt
// 中文: Autocomplete 輸入分類器薄殼 — 把原始 keystroke 序列分類為
// 中文: Hanzi / RomanWithTone / RomanNoTone(InputType)。實際分類邏輯已在 Rust
// 中文: lexicon::classify_input;此檔僅將 LexiconBridge.classifyInput 的結果取出 inputType。

package com.siansiansu.taigikeyboard.ime.text.composing

import com.siansiansu.taigikeyboard.engine.LexiconBridge
import com.siansiansu.taigikeyboard.ime.dictionary.InputType

/**
 * Classify raw autocomplete input into an [InputType] tier.
 *
 * Thin wrapper over [LexiconBridge.classifyInput]. Precedence contract:
 * `INVARIANT_LEX_INPUT_CLASSIFICATION_PRECEDENCE`.
 */
object AutocompleteInputClassifier {
    fun determineInputType(text: String): InputType = LexiconBridge.classifyInput(text).inputType
}
